File: app/src/main/java/com/selfmate/mes/selfmate/di/AppModule.java
package com.selfmate.mes.selfmate.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by vicki_mes on 11/18/2017.
 */

@Module
public class AppModule {
    private Application mApplication;

    public AppModule(Application application) {
        mApplication = application;
    }

    @Provides
    @Singleton
    Application providesApplication() {
        return mApplication;
    }
}

File: app/src/main/java/com/selfmate/mes/selfmate/di/AppModule.kt
package com.selfmate.mes.selfmate.di

import android.app.Application
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by vicki_mes on 11/18/2017.
 */

@Module
class AppModule(private val mApplication: Application) {

    @Provides
    @Singleton
    internal fun providesApplication(): Application = mApplication
}
